src/Angor/Client/Pages/Spend.razor

```diff
@@ -251,7 +251,7 @@
 
     List<(Transaction Transaction, string TrxId)> transactions = new();
 
-    Dictionary<string, string> selected = new();
+    Dictionary<int, HashSet<string>> selected = new();
```

Comment:
I see so basically here you are attempting to link a checkbox to a utxo, may I suggest using the utxo itself instead of the stage identifier and avoid the nasty hashset type.
In the `StageDataTrx` type you have both `Trxid` and `Outputindex` this is unique to the utxo, so I suggest to keep the dictionary as it was and use the `outpoint` "Trxid-Outputindex) as the key.